File: app/src/main/java/com/gooner10/popularmoviesapp/moviemain/MovieContract.java
package com.gooner10.popularmoviesapp.moviemain;

import com.gooner10.popularmoviesapp.data.MovieData;

import java.util.List;

/**
 * This specifies the contract between the Movie view and the Movie presenter.
 */
public interface MovieContract {
    interface View {
        void displayMovieList(List<MovieData> movieDataList);
    }

    interface UserActionsListener {
        void loadData();
    }
}

File: app/src/main/java/com/gooner10/popularmoviesapp/moviemain/MovieContract.kt
package com.gooner10.popularmoviesapp.moviemain

import com.gooner10.popularmoviesapp.data.MovieItem

/**
 * This specifies the contract between the Movie view and the Movie presenter.
 */
interface MovieContract {
    interface View {
        fun displayMovieList(movieDataList: List<MovieItem>)
    }
}
